src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -2487,21 +2491,44 @@ struct Gcx
 
                                 if (!pool.mark.test(biti))
                                 {
-                                    void* q = sentinel_add(p);
-                                    sentinel_Invariant(q);
-
-                                    if (pool.finals.nbits && pool.finals.test(biti))
-                                        rt_finalizeFromGC(q, sentinel_size(q, size), pool.getBits(biti));
-
-                                    assert(core.bitop.bt(toFree.ptr, i));
-
-                                    debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
-                                    leakDetector.log_free(sentinel_add(p));
-
-                                    debug (MEMSTOMP) memset(p, 0xF3, size);
+                                    ` ~ code ~ `
                                 }
                             }
-                        }
+                        }`;
+
+                        debug (SENTINEL)
```

Comment:
Please avoid the string mixin, it makes the code undebuggable.

I was rather thinking of:
```
debug(SENTINEL)
   enum doLoop = true;
else version(assert)
   enum doLoop = true;
else debug(...)
   ...
else
   bool doLoop = hasFinalizer;
if (doLoop)
````
followed by the existing code. If any of the debug conditions are enabled, peerformance is not a goal.

Reply:
Undebuggable how? You mean with an interactive debugger?

I wanted to avoid duplicating the condition and putting the other instance too far away from the first one, as they will inevitably go out of sync, as they already have in this very case.

Reply:
> Undebuggable how? You mean with an interactive debugger?

Yes.

BTW: I just noticed that `hasFinalizer` is already a defined a few lines above as I proposed. Maybe it is just missing the SENTINEL condition?

Reply:
> BTW: I just noticed that `hasFinalizer` is already defined a few lines above as I proposed.

Hah, well spotted.

OK, I'll change it to the old style. But, if this bitrots again...

Reply:
Thanks. No guarantees, though ;-)